sec1_intro_and_setup/test_app/config/storage.yml

```diff
@@ -0,0 +1,34 @@
+test:
```

Comment:
Whats this file for?

Reply:
This file allows you to specify the location to store your assets such as images.

Reply:
What do you mean by `location`?

Reply:
Cloud storage or local directory based on what you specify.

Reply:
Active Storage will read this file and knows where(the cloud storage or local directory you specified) to upload your assets to. 

Active Storage, the way Rails upload your assets to cloud storage by using 2 database tables to manage the assets you uploaded, 
- `active_storage_blobs` : to store the information about the assets you uploaded (filename, content type etc)
- `active_storage_attachments`:  to stores the relationship between your model and the file upload

---
Question:
1. Why is it called Active Storage? Where does it seem active?
2. Why does Active Storage use this method (creating database tables) to manage the assets? Reduce HTTP requests? Increase web loading speed? 
3. How do models differ from class? Why is it call model?
4. How does this file differ from the assets file? Is it that this file is to store what the users upload and the assets file stores what we need to show to the users on the webpage?

Reply:
All the questions you put here are my questions for you. I won't be answering here cause if you understood what active storage is and why it is used here, you would be able to answer these questions

Reply:
2. No clue.
3. Rails model seems to be often used in database and rails classes are is more towards object. Is there a superior relationship between models and class?
4. For now, I only understand that active storage is catered for clients to upload assets while the assets file is more related to static resources where the administrator/developers store assets needed for the app

Reply:
2. Because you would need to store the locataion of the file. Location here refers to URL, file name, etc.
3. A rails model is a representation of the database table. Given that it is a ruby class, it has some built in function to support the functionality of the database (eg: update, delete, where, save, etc). All ruby class that inherit from `ActiveRecord::Base` is taken as a model by rails. However, rails will only load all models within the `/model` folder by default. There is no one superior than the other. Its just a ruby class at the end that do a very specific thing
4. The purpose of this file is to provide a unified interface in dealing with file storage. When we store files, we are storing them outside of the database. What we store in the database is where the file is. So, where do we store them? Anywhere you want. It could be in AWS S3, google drive, dropbox, etc. Each of this place you want to store has their own API on how you can upload the file there by creating a new file, update the current file or delete the file. Active storage make it so that you don't have to worry about those integrations (you can just plug the ruby gem) and it will work. It is the same as database.yml. The same unified interface but allows you to connect to any db (postgres, mysql, mongodb, etc etc). This allows you to do: `model.save` and it will save the data regardless of what database you have. 

---

If you understand, please resolve this